addons/mrp_subcontracting_purchase/models/account_move_line.py

```diff
@@ -14,3 +14,24 @@ def _get_price_unit_val_dif_and_relevant_qty(self):
             components_cost -= sum(subcontract_production.move_raw_ids.stock_valuation_layer_ids.mapped('value'))
             price_unit_val_dif = price_unit_val_dif + components_cost / relevant_qty
         return price_unit_val_dif, relevant_qty
+
+    def _get_valued_in_moves(self):
+        res = super()._get_valued_in_moves()
+        # subcontracted move valuations are not linked to the PO move but its orig move (the MO finished move)
+        res |= res.filtered(lambda m: m.is_subcontract).move_orig_ids
+        return res
+
+    def _get_layer_price_unit(self, layer):
```

Comment:
you don't use `self` in the method. I would create this method on the stock valuation layer model instead